exec/java-exec/src/main/java/org/apache/drill/exec/store/enumerable/plan/VertexDrel.java

```diff
@@ -51,4 +58,15 @@ protected Object clone() {
   public LogicalOperator implement(DrillImplementor implementor) {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public @Nullable RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
+    double rowCount = estimateRowCount(mq);
+    double columnCount = Utilities.isStarQuery(getRowType()) ? STAR_COLUMN_COST : getRowType().getFieldCount();
+    double valueCount = rowCount * columnCount;
```

Comment:
Was it only queries with a pushed down join that were affected by there being no cost saving for adding a plugin projection?

Reply:
I think more queries were affected, but probably it was simpler to reproduce it on queries with joins.